### PR TITLE
feat: migrate the doc tool to docusaurus

### PR DESCRIPTION
**Pull request by iNorthIsle**
The PR purpose is to migrate doc tools to Docusaurus. it's a preview version, if you have any questions, please leave comments 🙇 

- [x] Migrating all resources to Docusaurus.
- [x] clean the unused files
- [x] verify the deploy flow
- [x] check all the references about Englua

Two issues:
- No content and images to fill the homepage and events pages
- Don't know how to display Engula's features on these pages

### CODE REVIEW

**Comment by tisonkun** [comment]
Thanks for preparing this PR @iNorthIsle! I'll check it out tomorrow ;-)

**Review by tisonkun — COMMENTED** [review]
A few quick comments here:

1. "Community" can be a top-level nav item. Links to forum, chatroom can be displayed also.
  * Live demo 0: https://www.docsy.dev/community/
  * Live demo 1: https://flink.apache.org/community.html
2. Docusaurus natively supports search bar, the draft doesn't turn it on, yet.
3. Can Docusaurus supports showing author and created time of a blog?
4. The main page should fulfill the screen. Below is a former demo - you don't have to follow it, of course, but fulfill the screen.

image

**Comment by tisonkun** [comment]
BTW, if you're familiar with Vercel and willing to show the deployment as you previously did (https://engula-website-48kxgtabm-inorthisle.vercel.app/), you can tell us how to configure the repo for supporting that if possible.

**Comment by iNorthIsle** [comment]
thanks for your check! 
will do 1,2,3 first base on the above comment, then re-design the section of the main page.

> BTW, if you're familiar with Vercel and willing to show the deployment as you previously did (https://engula-website-48kxgtabm-inorthisle.vercel.app/), you can tell us how to configure the repo for supporting that if possible.

Vercel will auto-detect what`s kind of the framework preset and the scripts, what's you need to do is import the repo, and then deploy it.
image

**Comment by iNorthIsle** [comment]
hi, @tisonkun has pushed some updates about 1,2,3 in the above comment, replaced the https://engula-website-48kxgtabm-inorthisle.vercel.app/ by https://engula-github-io-euroe1ya4-inorthisle.vercel.app/, so you can preview the changes about engula.github.io

**Comment by tisonkun** [comment]
@iNorthIsle thanks! I'll review it in this weekend :P

**Comment by iNorthIsle** [comment]
> @iNorthIsle thanks! I'll review it in this weekend :P

ok, thanks! In the meantime, I'll re-design the section of the main page and make the SVG animation. 🙇

File: docusaurus.config.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        },
+        items: [
+          {
+            type: 'doc',
+            docId: 'intro',
+            position: 'right',
+            label: 'Docs',
+          },
+          { to: '/event', label: 'Event', position: 'right' },
+          { to: '/blog', label: 'Blog', position: 'right' },
```
Comment (tisonkun) [inline docusaurus.config.js:66]:
```suggestion
          { to: '/event', label: 'Events', position: 'right' },
          { to: '/blog', label: 'Blogs', position: 'right' },
```

ditto related identifiers

File: docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      },
+      prism: {
+        theme: lightCodeTheme,
+        darkTheme: darkCodeTheme,
+      },
+      algolia: {
+        // If Algolia did not provide you any appId, use 'BH4D9OD16A'
+        appId: 'YOUR_APP_ID',
+        // Public API key: it is safe to commit it
+        apiKey: 'YOUR_SEARCH_API_KEY',
```
Comment (tisonkun) [inline docusaurus.config.js:135]: It seems algolia searching mechanism is broken now.

**Review by tisonkun — COMMENTED** [review]
Following comments:

For events page, you can populate the release of 0.2 as the content like https://skywalking.apache.org/events/release-apache-skywalking-kong-0-2-0 .

For docs page, you can replace the manifesto with our overall design https://github.com/engula/engula/blob/main/docs/design.md.

Other comments inline.

**Comment by tisonkun** [comment]
@iNorthIsle do you have a plan on pushing forward this pull request? Especially due to the coming Spring Festival, an estimate can help us to align on the progress :)

**Comment by iNorthIsle** [comment]
> @iNorthIsle do you have a plan on pushing forward this pull request? Especially due to the coming Spring Festival, an estimate can help us to align on the progress :)

sorry about I haven't updated anything 😣 , my personal 💻 broke earlier this week, I got it back today and will keep going on to align on the progress 🙇

**Comment by iNorthIsle** [comment]
hi, @tisonkun, has pushed some updates:
- re-design homepage section (without the SVG animation, In fact, I don't know how to use the animation to show what's Engula should display)
- auto fetching the release notes of Engula and displaying them in the **events** page
- replaced the manifesto with overall design

Please help to check it on https://engula-github-io.vercel.app/  🙇 

> It seems algolia searching mechanism is broken now.

💡  The Algolia service need to create the account by U self

**Comment by tisonkun** [comment]
@iNorthIsle thanks! I'll review in this week. For search engine, is it possible to use a local search solution for now?

**Comment by iNorthIsle** [comment]
> @iNorthIsle thanks! I'll review in this week. For search engine, is it possible to use a local search solution for now?

ok, will update the PR for your check before tomorrow

**Comment by iNorthIsle** [comment]
hi, @tisonkun has pushed the updates of local search, U can check it on https://engula-github-io.vercel.app/
image

**Review by tisonkun — COMMENTED** [review]
Thanks for your updates!

Several comments below. I think we're in the right direction now.

1. You can leave only the "Design" under docs. As this is the first PR for website framework, we can further add new content without messing up things in one PR.
2. All blogs' author is @huachaohuang "Richard".
3. Main page blocks can be squashed:
    1. "Engula Design Goal" => "Design Goal"
    2. Events: Sync with release 0.2 or just remove the block.
    3. Keep in touch: removed
    4. Try Engula now: removed
    5. Main view: "Get Started" => "Design" and "GitHub" Button, see also live demo shared above.

**Comment by iNorthIsle** [comment]
@tisonkun pushed the squashed updates base on the above comments 🙇

**Comment by tisonkun** [comment]
@iNorthIsle thanks for your work over the time!

@huachaohuang I think this PR is ready for review and you can see the result at https://engula-github-io.vercel.app.

It provides all functionality covered by current framework and is easy to extend.

Also, to deploy the website, we can make use of Vercel as described at https://github.com/engula/engula.github.io/pull/21#issuecomment-1009546772, which is the de-facto front-end workflow and we can preview any incoming pull request then.

If you have any question or concern, please leave a (review) comment and @iNorthIsle and I should be glad to answer them.

**Comment by huachaohuang** [comment]
@iNorthIsle @tisonkun Thanks for your great work here! Sorry that I didn't pay attention here until now. Mostly because I don't know these things very much. The overall website looks good to me. But I still need something to figure out how things work. I will do that in these two days. Meanwhile, I think it's OK to proceed if there is anything left to be done.

**Comment by huachaohuang** [comment]
BTW, the logo in the left corner does need some improvement. Maybe I can draw a new one later :)

**Comment by huachaohuang** [comment]
It seems that there is no syntax highlight for code blocks (e.g., https://engula-github-io.vercel.app/blog/tutorial-0.2#hash-engine). Is that expected and can we support it?

**Comment by iNorthIsle** [comment]
Hi, @huachaohuang ,it can add highlight support for the code block, and I have pushed the updates, U can check it by visiting the above link. cc @tisonkun 🙇

File: blog/demo-1.md [hunk 3]
```diff
@@ -13,7 +15,7 @@ We then describe the experiments we conducted on the demo and share some lessons
 
 The architecture of the demo is as follow:
 
-![Architecture](../images/demo-1-architecture.drawio.svg)
+![Architecture]/img/demo-1-architecture.drawio.svg)
```
Comment (huachaohuang) [inline blog/demo-1.md:18]:
```suggestion
![Architecture](/img/demo-1-architecture.drawio.svg)
```

File: blog/tutorial-0.2.md [hunk 4]
```diff
@@ -62,7 +64,7 @@ async fn main() -> Result<()> {
 
 Now, you can run the example with:
 
-```sh
+```js
```
Comment (huachaohuang) [inline blog/tutorial-0.2.md:67]: This is a shell command, not `js`.

File: docusaurus.config.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        },
+        items: [
+          {
+            type: 'doc',
+            docId: 'design',
+            position: 'right',
+            label: 'Docs',
+          },
+          { to: '/events', label: 'Events', position: 'right' },
+          { to: '/blog', label: 'Blogs', position: 'right' },
```
Comment (huachaohuang) [inline docusaurus.config.js:66]:
```suggestion
          { to: '/blog', label: 'Blog', position: 'right' },
```

**Review by huachaohuang — COMMENTED** [review]
Some small fixes.

**Comment by huachaohuang** [comment]
I think I basically understand the new website now. The current setup looks good to me, although we will need to update some content to reflect the new design of Engula later. @iNorthIsle @tisonkun what's the next move we should take here?

**Comment by tisonkun** [comment]
I don't know whether or not docunasour can be generated statically and served as GitHub Pages. But we can deploy the site via Vercel and resolve engula.io to the provided Vercel app url. See also this guide: https://vercel.com/guides/deploying-docusaurus-with-vercel For proceeding the actions, it requires permission to install app on repos.

@iNorthIsle it seems in your branch you don't remove the mkdocs deployment workflow nor introduce a docunasour deployment workflow. What's your preference on deployment?

**Comment by huachaohuang** [comment]
I need some confirmation before merging this:

- Is there anything left to be done here? (We can update the content later)
- After merging, all I need to do next is to deploy it with Vercel?

**Comment by iNorthIsle** [comment]
> Is there anything left to be done here? (We can update the content later)

I found there left some responsive issues, will finish it before tomorrow.

> After merging, all I need to do next is to deploy it with Vercel?

YES

**Comment by huachaohuang** [comment]
@iNorthIsle thanks again. I am going to merge this and try to deploy it now.

**Comment by huachaohuang** [comment]
Great, I think the website is online! @iNorthIsle you should consider adding yourself to the [medal wall](https://www.engula.io/docs/general/community) 😄

File: src/pages/index.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      </div>
+    </header>
+  );
+}
+
+export default function Home(): JSX.Element {
+  const { siteConfig } = useDocusaurusContext();
+  return (
+    <Layout
+      title={`Hello from ${siteConfig.title}`}
```
Comment (zojw) [inline src/pages/index.tsx:43]: image

Is it a desired title~?(browser's tab name 😄

Comment (huachaohuang) [reply src/pages/index.tsx:43]: Nice catch. We get a lot of things to update before release 0.3.

**Pull request merged**